### PR TITLE
fix(upgrade): don't create BE if up to date

### PR DESCRIPTION
**Pull request by ELDiablO59152**
### Fixed
- [UPGRADE-OS] Wrongly create new BE when already up to date
- [UPGRADE-OS] Incorrect date format
- [UPGRADE-OS] Correctly Download and use the upgrade when upgrading only one or more jails and base is up-to-date

### CODE REVIEW

File: usr/local/share/vulture-utils/upgrade-os.sh [hunk 1]
```diff
@@ -74,11 +74,8 @@ update_system() {
             _options="${_options} -v $system_version -U"
         fi
         if [ $snapshot_system -gt 0 ]; then
-            /sbin/bectl create "${_snap_name}" || finalize 1 "Could not create a new Boot Environment!"
             clean_old_BEs "$keep_previous_snap"
```
Comment (frikilax) [inline usr/local/share/vulture-utils/upgrade-os.sh]:
```suggestion
```
This should not be done here unconditionaly

File: usr/local/share/vulture-utils/upgrade-os.sh [hunk 2]
```diff
@@ -164,7 +149,7 @@ finalize() {
     has_pending_BE
```
Comment (frikilax) [inline usr/local/share/vulture-utils/upgrade-os.sh]:
```suggestion
    if has_pending_BE; then
        clean_old_BEs "$keep_previous_snap"
    fi
```

File: usr/local/share/vulture-utils/upgrade-os.sh (hunk 1 above)
Comment (ELDiablO59152) [reply usr/local/share/vulture-utils/upgrade-os.sh]: Done

File: usr/local/share/vulture-utils/upgrade-os.sh (hunk 2 above)
Comment (ELDiablO59152) [reply usr/local/share/vulture-utils/upgrade-os.sh]: Suggestion applied

**Review by frikilax — APPROVED** [review]

**Pull request merged**